backend/src/services/groups/groups.service.ts

```diff
@@ -0,0 +1,46 @@
+import { GroupsRequestDto, GroupsResponseDto } from '~/common/types/types';
+import {
+  permission as permissionServ,
+  groupsToPermissions as groupsToPermissionsServ,
+} from '~/services/services';
+import { groups as groupsRep } from '~/data/repositories/repositories';
+
+type Constructor = {
+  groupsRepository: typeof groupsRep;
+  permissionService: typeof permissionServ;
+  groupsToPermissionsService: typeof groupsToPermissionsServ;
+};
+
+class Groups {
+  #groupsRepository: typeof groupsRep;
+  #permissionService: typeof permissionServ;
+  #groupsToPermissionsService: typeof groupsToPermissionsServ;
+
+  constructor({
+    groupsRepository,
+    permissionService,
+    groupsToPermissionsService,
+  }: Constructor) {
+    this.#groupsRepository = groupsRepository;
+    this.#permissionService = permissionService;
+    this.#groupsToPermissionsService = groupsToPermissionsService;
+  }
+
+  async createGroup(
+    groupsRequestDto: GroupsRequestDto,
+  ): Promise<GroupsResponseDto> {
+    const { permissionIds } = groupsRequestDto;
+    const group = await this.#groupsRepository.create(groupsRequestDto);
+
+    for (const id of permissionIds) {
+      await this.#groupsToPermissionsService.createGroupsToPermissions({
+        groupId: group.id,
+        permissionId: id,
+      });
```

Comment:
```suggestion
      await Promisse.all(permissionIds.map(it => this.#groupsToPermissionsService.createGroupsToPermissions({
        groupId: group.id,
        permissionId: it,
      }));
```

Reply:
Earlier I used Promise.all, but mentor @v0ldemar01 explained to me why it's a bad practice to use Promise.all and gave advice to use "for await approach", so, now I'm a little bit confused, what is better and who is right

Reply:
lets us to discuss it. Will be back in a few. Resole other comments for now

Reply:
We discussed it. Do with promisse.all. Sorry for confusion

Reply:
So sorry, please add current suggestion)

Reply:
Could you explain to me please, why it's better to use promise.all here?

Reply:
`for await ... of` will process async requests one by one,, `Promise.all` with mapped request Promises will process them concurrently